File: Python Connection/redis-connection.py
import redis

REDIS_PASS = "abc@123"
PORT = 6380
REDIS_URL = f"redis://localhost:{REDIS_PASS}@myredis:{PORT}"

def connect_to_redis():
    try:
        # Can also connect using a specific host, URL, and other details. ↓
        client = redis.StrictRedis(host="localhost", port=PORT, password=REDIS_PASS)
        # client = redis.from_url(REDIS_URL)
        print("Connected to Redis server!")
        return client
    except redis.exceptions.ConnectionError as err:
        print(f"Connection error: {err}")
        return None

redis_connection = connect_to_redis() 

if redis_connection:
    print(f"Connection Details: {redis_connection} \n")
    redis_connection.set("Day", "Friday", ex=1500)
    value = redis_connection.get("Day")
    print(f"Retrieved value: {value.decode('utf-8')}")
    
    # # Close the connection (optional)
    redis_connection.close()
    print("Connection closed.")

File: JS Connection/redis-connection.js
const PORT = 6380;
const redisUrl = `redis://localhost:${PORT}`;

const redis = require('redis');

async function main() {
    const client = redis.createClient({
        url: redisUrl,
        password: 'abc@123'
    });
    

    client.on('error', (err) => {
        console.error('Redis Client Error:', err);
        throw new Error(`Redis Server is not available on port: ${PORT}`);
    });

    try {
        await client.connect();
        console.log("Connected to Redis server!");
        console.log("Connection Details");
        console.table(client);
    } catch (err) {
        console.error('Failed to connect to Redis:', err);
        return;
    }

    try {
        await client.set('Month', 'June', {
            EX: 1500
        });
        const value = await client.get('Month');
        console.log('Month:', value);
    } catch (err) {
        console.error('Error performing Redis operations:', err);
    } finally {
        await client.quit();
        console.log("Connection Closed");
    }
}

main().catch(err => {
    console.error('Unexpected error:', err);
    process.exit(1);
});